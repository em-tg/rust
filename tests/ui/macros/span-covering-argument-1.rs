//@ check-pass
macro_rules! bad {
    ($s:ident whatever) => {
        {
            let $s = 0;
            *&mut $s = 0;
            //~^ WARNING cannot borrow `foo` as mutable, as it is not declared as mutable [E0596]
        }
    }
}

fn main() {
    bad!(foo whatever);
}
