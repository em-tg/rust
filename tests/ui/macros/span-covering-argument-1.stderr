warning[E0596]: cannot borrow `foo` as mutable, as it is not declared as mutable
  --> $DIR/span-covering-argument-1.rs:6:14
   |
LL |             *&mut $s = 0;
   |              ^^^^^^^ cannot borrow as mutable
...
LL |     bad!(foo whatever);
   |     ------------------ in this macro invocation
   |
   = note: `#[warn(mut_non_mut)]` on by default
   = note: this warning originates in the macro `bad` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider changing this to be mutable
   |
LL |             let mut $s = 0;
   |                 +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
